### PR TITLE
Bigafdb

### PR DESCRIPTION
**Pull request by rmk65**
Updated with new command line options and set to version 0.3.5. New options include:
"pdb_local" - set a local PDB mirror
"use_api" -  use EBI phmmer search API (False by default)
"max_hits" - maximum number of models to prepare from each search 
"afdb_seqdb" - set path to local afdb sequence listing file (e.g. full fasta list for afdb)
"nproc" - number of CPU cores to give to Phmmer

### CODE REVIEW

File: mrparse/mr_alphafold.py [hunk 1]
```diff
@@ -156,7 +156,8 @@ def prepare_pdb(hit, plddt_cutoff, database_version):
     trucate to required residues
     calculate the MW
     """
-    pdb_name = f"AF-{hit.pdb_id}-F1-model_{database_version}.pdb"
+    #pdb_name = f"AF-{hit.pdb_id}-F1-model_{database_version}.pdb"
```
Comment (hlasimpk) [inline mrparse/mr_alphafold.py:162]: We can remove this

File: mrparse/mr_hit.py [hunk 2]
```diff
@@ -99,17 +99,21 @@ def __str__(self):
         return out_str
 
 
-def find_hits(seq_info, search_engine=PHMMER, hhsearch_exe=None, hhsearch_db=None, afdb_seqdb=None, phmmer_dblvl=95):
+def find_hits(seq_info, search_engine=PHMMER, hhsearch_exe=None, hhsearch_db=None, afdb_seqdb=None, phmmer_dblvl=95, localrun=False, max_hits=10):
```
Comment (hlasimpk) [inline mrparse/mr_hit.py]: For consistency, can we rename localrun to local_run?

File: mrparse/mr_hit.py [hunk 3]
```diff
@@ -118,72 +122,121 @@ def find_hits(seq_info, search_engine=PHMMER, hhsearch_exe=None, hhsearch_db=Non
...
-        for hsp in hit.hsps:
+    if af2 or searchio_type == "hmmer3-text":
+        if af2:
+            fix_af_phmmer_log(logfile, "phmmer_af2_fixed.log")
+            logfile = "phmmer_af2_fixed.log"
+    
+        # Read logfile with searchDB
+        from mrparse.searchDB import Phmmer  
+    
+        plog=open(logfile, "r")
```
Comment (hlasimpk) [inline mrparse/mr_hit.py:156]: This section could do with reformating to better fit the other code in MrParse. Basically just removing capitals from variable names, including underscores between words and adding spaces before/after = when assigning variables.

File: mrparse/mr_hit.py [hunk 4]
```diff
@@ -118,72 +122,121 @@ def find_hits(seq_info, search_engine=PHMMER, hhsearch_exe=None, hhsearch_db=Non
...
-            hitDict[hit_name] = sh
+                sh.name = hit_name
+                if sh.rank <= max_hits:
+                    hitDict[hit_name] = sh
+
     return hitDict
 
 
-def _find_json_hits(json_file, target_sequence):
+def _find_json_hits(json_file, target_sequence, max_hits=10):
```
Comment (hlasimpk) [inline mrparse/mr_hit.py:257]: max hits should feed in to the json search elsewhere, in `run_phmmer_alphafold_api` there is a range `1,10` that gets parsed into the API. This returns 10 hits by default. You might want to set max hits here,

File: mrparse/searchDB.py [hunk 5]
```diff
@@ -0,0 +1,433 @@
+#! /usr/bin/env ccp4-python
```
Comment (hlasimpk) [inline mrparse/searchDB.py:1]: I might be inclined to change the Camel case here to snake case too, although I understand it's a lot of work! There are also a few lines throughout this that are commented out, are these needed?

**Review by hlasimpk — CHANGES_REQUESTED** [review]
Just small formatting things.

File: mrparse/mr_hit.py (hunk 4 above)
Comment (rmk65) [reply mrparse/mr_hit.py:257]: Added this

File: mrparse/mr_hit.py (hunk 3 above)
Comment (rmk65) [reply mrparse/mr_hit.py:156]: Made this change.

File: mrparse/mr_alphafold.py (hunk 1 above)
Comment (rmk65) [reply mrparse/mr_alphafold.py:162]: Removed

**Review by hlasimpk — APPROVED** [review]

**Pull request merged**